### DISCUSSION TITLE
Add a way to get a member as IPublishedContent without it being async

### DISCUSSION THREAD

**Discussion by jemayn** [umbraco/Umbraco-CMS#13945] [Features and ideas]
I am trying to extend the member index with some of our custom member fields, and when doing that I am hooking into the TransformingIndexValues event where I'd like to get the corresponding IPublishedContent model for the member based on the id.

Back in v8 this would be super easy to do as we could just use the UmbracoHelper to [get the member by id](https://our.umbraco.com/documentation/Reference/Querying/UmbracoHelper/#member1234).

However, it seems in the later versions this has been removed and moved into the `IMemberManager` which then requires you to get the member by id, and then transform it from a `MemberIdentityUser` to `IPublishedContent` like this:

```csharp
@{
    MemberIdentityUser memberById = await _memberManager.FindByEmailAsync("test@member.com");
    IPublishedContent memberAsContent = _memberManager.AsPublishedMember(memberById);
}
```

I don't know the underlying reasons for the change, and whether being able to get the IPC members in a sync way is possible with the way members work now. But it would make some things so much easier if it was possible to do so!

**Comment by bergmania** [comment]
Hi @jemayn.

Do you need to `MemberIdentityUser`? If `IMember` has enough data you can use `IMemberService.GetById`/ `IMemberService.GetByEmail`

**Reply by jemayn** [reply]
My specific usecase is that I need to run a scheduled job that will send out email notifications to members that have a specific property set.

My thought was to hook into the member index and ensure that property is indexed (as the member index unlike the content index only indexes specific default fields). Then on the job I could do a search for the value via the index and then get a list of members to send the email to.

So it would be possible to do it via the member service, but then on a full reindex of the member index I'd hit the memberservice for each record to get the user and their properties - to ensure a better performance I'd much rather do an IPublishedContent lookup - but I may have to use the memberService for now..

Otherwise I may need to set the email subscribtion settings in an external db table or something - but it would be super nice to be able to handle it all within Umbraco 🙂

**Reply by bergmania** [reply]
Members is not - and have never been - saved in NuCache, so whatever translates to IPublishedContent still needs the data from an `IMember`. This is also what happens in `memberManager.AsPublishedMember(memberById)`.

**Reply by jemayn** [reply]
Ahh that's good to know! 

I just assumed it was due to the model. I will try with the memberService or find a different way around the issue then - thanks for the help Bjarke!
